compiler/src/dotty/tools/dotc/core/tasty/PositionPickler.scala

```diff
@@ -19,9 +19,18 @@ import util.Spans._
 class PositionPickler(
     pickler: TastyPickler,
     addrOfTree: PositionPickler.TreeToAddr,
-    treeAnnots: untpd.MemberDef => List[tpd.Tree]) {
+    treeAnnots: untpd.MemberDef => List[tpd.Tree],
+    relativePathReference: String){
+
+  def this(
+    pickler: TastyPickler,
+    addrOfTree: PositionPickler.TreeToAddr,
+    treeAnnots: untpd.MemberDef => List[tpd.Tree]
+  )(using ctx: Context) =
+    this(pickler, addrOfTree, treeAnnots, ctx.settings.sourceroot.value)
```

Comment:
The point is to not use `Context` in this file, we do not want to reintroduce this dependency that was a pain to remove in the first place. Why not just pass `ctx.settings.sourceroot.value` directly?

```scala
new PositionPickler(...,..., ctx.settings.sourceroot.value)
```
This would only need 2 lines changed rather than the addition of a secondary constructor.
* https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/transform/Pickler.scala#L75
* https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala#L170

Reply:
> The point is to not use Context in this file, we do not want to reintroduce this dependency that was a pain to remove in the first place. Why not just pass ctx.settings.sourceroot.value directly?

I think using it in secondary constructor achieves several goals: 

(1) zero-dependence inside the class (easy to remove)
(2) DRY
(3) encapsulation

However, if the 2nd and 3rd are not the goals, I can change the code as you suggested.